pkg/services/apiserver/auth/authenticator/signedinuser.go

```diff
@@ -0,0 +1,42 @@
+package authenticator
+
+import (
+	"net/http"
+	"strconv"
+
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	k8suser "k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/klog/v2"
+
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+)
+
+var _ authenticator.RequestFunc = signedInUserAuthenticator
+
+func signedInUserAuthenticator(req *http.Request) (*authenticator.Response, bool, error) {
+	ctx := req.Context()
+	signedInUser, err := appcontext.User(ctx)
+	if err != nil {
+		klog.V(5).Info("failed to get signed in user", "err", err)
+		return nil, false, nil
+	}
+
+	userInfo := &k8suser.DefaultInfo{
+		Name:   signedInUser.Login,
+		UID:    strconv.FormatInt(signedInUser.UserID, 10),
```

Comment:
this is a change from the last commit. not sure if user ID makes sense here or in the name. thoughts?

Reply:
It's an interesting question, we have the new uid column which is what will become the Name in k8s, but we don't have the k8s uid.

Looking at https://pkg.go.dev/k8s.io/apiserver/pkg/authentication/user#Info it seems like these aren't intended to map directly to the name and uid of a k8s resource, so we could likely stick with login for the name and use the new uid column for the uid.

Similarly for the groups we should probably use the new uid column?

Realistically the intent is to not trust these headers and instead use the verified id token, but that would at least give us a better chance of a globally-unique identifier vs the autoincrement ids